lib/IO/String.pm

```diff
@@ -36,10 +36,48 @@ TODO
 
 #| An L<IO::Handle> implementation that writes to memory.
 class IO::String:ver<0.1.0>:auth<hoelzro> is IO::Handle {
-    has @.contents;
+    has Str $.buffer = '';
+    has Int $.pos = 0;
+
+    multi method open(IO::String:D:) {
+        my Str $new-buffer = '';
+        $!buffer := $new-buffer;
+        $!pos = 0;
+    }
+
+    multi method open(IO::String:D: Str $buffer is rw) {
+        $!buffer := $buffer;
```

Comment:
This "read-write" buffer being passed in kind of bugs me a bit;  if I understand this correctly, that means that this code will alter `$s`, right?

```perl6
my $s = "foobar\n";
my $io = IO::String.open($s);
$io.say: "hello";
```